get_smb_mount.go

```diff
@@ -0,0 +1,103 @@
+package getter
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// SmbMountGetter is a Getter implementation that will download a module from
+// a shared folder using the file system using FileGetter implementation.
+// For Unix and MacOS users, the Getter will look for usual system specific mount paths such as:
+// /Volumes/ for MacOS
+// /run/user/1000/gvfs/smb-share:server=<hostIP>,share=<path> for Unix
+type SmbMountGetter struct {
+}
+
+func (g *SmbMountGetter) Mode(ctx context.Context, u *url.URL) (Mode, error) {
+	if u.Host == "" || u.Path == "" {
+		return 0, new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(u)
+	u.Path = prefix + path
+
+	return new(FileGetter).Mode(ctx, u)
+}
+
+func (g *SmbMountGetter) Get(ctx context.Context, req *Request) error {
+	if req.u.Host == "" || req.u.Path == "" {
+		return new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(req.u)
+	req.u.Path = prefix + path
+
+	return new(FileGetter).Get(ctx, req)
+}
+
+func (g *SmbMountGetter) GetFile(ctx context.Context, req *Request) error {
+	if req.u.Host == "" || req.u.Path == "" {
+		return new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(req.u)
+	req.u.Path = prefix + path
+
+	return new(FileGetter).GetFile(ctx, req)
+}
+
+func (g *SmbMountGetter) findPrefixAndPath(u *url.URL) (string, string) {
+	var prefix, path string
+	switch runtime.GOOS {
+	case "windows":
+		prefix = string(os.PathSeparator) + string(os.PathSeparator)
+		path = filepath.Join(u.Host, u.Path)
+	case "darwin":
+		prefix = string(os.PathSeparator)
+		path = filepath.Join("Volumes", u.Path)
+	case "linux":
+		prefix = string(os.PathSeparator)
+		path = fmt.Sprintf("run/user/1000/gvfs/smb-share:server=%s,share=%s", u.Host, strings.TrimPrefix(u.Path, prefix))
```

Comment:
Not sure about this one. I'm thinking maybe Linux user should always have smbclient installed for this to work.

Reply:
Should this be something like `run/user/$uid/gvfs` in case the user is not 1000 ?